crates/pathfinder/src/state/block_hash.rs

```diff
@@ -0,0 +1,310 @@
+use anyhow::{Context, Error, Result};
+use bitvec::prelude::BitView;
+use stark_hash::{stark_hash, StarkHash};
+
+use crate::core::{SequencerAddress, StarknetBlockHash};
+use crate::sequencer::reply::{
+    transaction::{Event, Receipt, Transaction},
+    Block,
+};
+use crate::state::merkle_tree::MerkleTree;
+
+pub fn compute_block_hash(block: &Block) -> Result<StarknetBlockHash> {
+    let transaction_commitment = calculate_transaction_commitment(&block.transactions)?;
+    let event_commitment = calculate_event_commitment(&block.transaction_receipts)?;
+
+    anyhow::ensure!(block.block_number.is_some());
+    let block_number = block.block_number.unwrap();
+    anyhow::ensure!(block.state_root.is_some());
+    let state_root = block.state_root.unwrap();
+
+    let num_transactions: u64 = block.transactions.len().try_into()?;
```

Comment:
I would prefer expect for these, the numerrorwhatever which comes out of usize->u64 will probably be very confusing to see in logs. Also, since this is converting an usize of an Vec if I understand correctly, we would hope it's under u64::MAX, or we have some redesigning to do :)

Same applies to num_events.